beacon-chain/p2p/pubsub_filter.go

```diff
@@ -0,0 +1,142 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	pubsubpb "github.com/libp2p/go-libp2p-pubsub/pb"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/feed"
+	statefeed "github.com/prysmaticlabs/prysm/beacon-chain/core/feed/state"
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p/encoder"
+	"github.com/prysmaticlabs/prysm/shared/p2putils"
+)
+
+var _ pubsub.SubscriptionFilter = (*subscriptionFilter)(nil)
+
+const pubsubSubscriptionRequestLimit = 100
+
+// subscriptionFilter handles filtering pubsub topic subscription metadata from peers.
+// Notice: This filter does not support phase 1 fork schedule yet.
+type subscriptionFilter struct {
+	ctx      context.Context
+	notifier statefeed.Notifier
+
+	// Once the genesis state is initialized, this filter is now aware of the genesis fork digest.
+	// This filter does not check the fork schedule for planned forks and their respective digests.
+	initialized        bool
+	currentForkDigest  string
```

Comment:
Is it possible to retrieve the fork digest from our parent `p2p` service ? I would rather track fork digest in one place if possible.